test/common/router/router_test.cc

```diff
@@ -4125,48 +4134,41 @@ TEST(RouterFilterUtilityTest, SetUpstreamScheme) {
 
 TEST(RouterFilterUtilityTest, ShouldShadow) {
   {
-    TestShadowPolicy policy;
+    ShadowPolicyPtr policy = std::make_unique<TestShadowPolicy>();
     NiceMock<Runtime::MockLoader> runtime;
     EXPECT_CALL(runtime.snapshot_, featureEnabled(_, _, _, _)).Times(0);
-    EXPECT_FALSE(FilterUtility::shouldShadow(policy, runtime, 5));
+    EXPECT_FALSE(FilterUtility::shouldShadow(*policy, runtime, 5));
```

Comment:
also realized there's no need for these to be unique_ptrs since the interface was reverted to const-ref. Will update these.